src/main/java/appland/telemetry/TelemetryService.java

```diff
@@ -0,0 +1,67 @@
+package appland.telemetry;
+
+import com.intellij.notification.NotificationType;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.project.Project;
+
+import appland.AppMapBundle;
+import appland.notifications.AppMapNotifications;
+import appland.settings.AppMapApplicationSettingsService;
+import appland.telemetry.data.BaseData;
+import appland.telemetry.data.TelemetryEvent;
+
+
+import org.jetbrains.annotations.NotNull;
+
+public class TelemetryService {
+    @NotNull protected final String instrumentationKey = "f688d11d-047a-48e0-87cb-fbdf38ecf697";
+    @NotNull protected final String ingestionEndpoint = "https://centralus-2.in.applicationinsights.azure.com";
+    @NotNull protected final AppInsightsClient client = new AppInsightsClient(ingestionEndpoint);
+
+    public static @NotNull TelemetryService getInstance() {
+        return ApplicationManager.getApplication().getService(TelemetryService.class);
+    }
+
+    public interface TelemetryEventBuilder {
+        BaseData build(@NotNull BaseData eventData);
+    }
+
+    public boolean isEnabled() {
+        return AppMapApplicationSettingsService.getInstance().isTelemetryEnabled();
```

Comment:
Do you have to check the env variable `APPMAP_TELEMETRY_DISABLED` here?

Reply:
A native setting is be preferred here. In the future we can use this flag to determine whether or not `APPMAP_TELEMETRY_DISABLED` is passed to child processes. However, none of the commands we're using at the moment will even emit telemetry so this will be a no op.